xbmc/cores/VideoPlayer/Buffers/VideoBufferDMA.cpp

```diff
@@ -119,20 +119,29 @@ bool CVideoBufferDMA::Alloc()
 
 void CVideoBufferDMA::Export(AVFrame* frame, uint32_t width, uint32_t height)
 {
-  m_planes = av_pix_fmt_count_planes(static_cast<AVPixelFormat>(frame->format));
+  AVPixelFormat pix_fmt = static_cast<AVPixelFormat>(frame->format);
+  m_planes = av_pix_fmt_count_planes(pix_fmt);
+  int h_shift, v_shift;
+
+  if (av_pix_fmt_get_chroma_sub_sample(pix_fmt, &h_shift, &v_shift))
+    throw std::runtime_error(
```

Comment:
Is a failure in an external library really something we want to hard crash to desktop for?